agent/lib/kontena/workers/service_pod_manager.rb

```diff
@@ -0,0 +1,131 @@
+require_relative 'service_pod_worker'
+require_relative '../models/service_pod'
+
+module Kontena::Workers
+  class ServicePodManager
+    include Celluloid
+    include Celluloid::Notifications
+    include Kontena::Logging
+    include Kontena::Helpers::RpcHelper
+
+    attr_reader :workers
+
+    trap_exit :on_worker_exit
+    finalizer :finalize
+
+    def initialize(autostart = true)
+      @workers = {}
+      subscribe('agent:node_info', :on_node_info)
+      subscribe('service_pod:update', :on_update_notify)
+      async.start if autostart
+    end
+
+    def start
+      node # blocks until it's available
+      populate_workers_from_docker
+      loop do
+        populate_workers_from_master
+        sleep 30
+      end
+    end
+
+    # @return [Node]
+    def node
+      while @node.nil?
+        @node = Actor[:node_info_worker].node
```

Comment:
I agree, maybe we should remove `Actor[:node_info_worker].node` call and just wait until `@node` is available (this logic was ok before I restored `:on_node_info`)